### PR TITLE
boards: nxp: Add frdm_rw612 board

### PR DESCRIPTION
**Pull request by decsny**
Add basic board definition of the freedom rw612 board.

### CODE REVIEW

**Review by dleach02 — DISMISSED** [review]

File: boards/nxp/frdm_rw612/frdm_rw612_defconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright 2022 NXP
```
Comment (danieldegrasse) [inline boards/nxp/frdm_rw612/frdm_rw612_defconfig:2]: Is this copyright accurate? I guess the file is copied from the RD-RW612-BGA, which is copyright 2022?

File: boards/nxp/frdm_rw612/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
++-----------+------------+-----------------------------------+
+| GPIO      | on-chip    | gpio                              |
++-----------+------------+-----------------------------------+
+| USART     | on-chip    | serial                            |
++-----------+------------+-----------------------------------+
+
+
+The default configuration can be found in the defconfig file:
+
+   :zephyr_file:`boards/nxp/frdm_rw612/frdm_rw612/`
```
Comment (danieldegrasse) [inline boards/nxp/frdm_rw612/doc/index.rst:46]: Path here needs to be updated

File: boards/nxp/frdm_rw612/frdm_rw612_defconfig (hunk 1 above)
Comment (decsny) [reply boards/nxp/frdm_rw612/frdm_rw612_defconfig:2]: yes

File: boards/nxp/frdm_rw612/doc/index.rst (hunk 2 above)
Comment (decsny) [reply boards/nxp/frdm_rw612/doc/index.rst:46]: i think this is a typo, this is a strange path

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**